detekt/defs.bzl

```diff
@@ -50,6 +50,9 @@ def _impl(ctx):
         action_outputs.append(xml_report)
         detekt_arguments.add("--report", "xml:{}".format(xml_report.path))
 
+    if ctx.attr.autocorrect:
+        detekt_arguments.add("--auto-correct")
```

Comment:
Have a weird experience with sandboxing here.

* If I declare inputs as outputs (since source files are mutated) — Bazel crashes on me with `java.lang.IllegalStateException: Source in outputs`. This is reasonable, inputs should not be changed to avoid recursive changes.
* If I change the integration test target `detekt_without_config_with_baseline_with_plugin` to enable autocorrect and run `bazel build` on it — it runs fine, even with the `--worker_sandboxing` Bazel argument.
* If I change the same target but change `supports-workers` to `0` — Bazel throws an error `Operation not permitted` since the sandbox is in place. But shouldn’t it break with worker sandbox as well?

Since the autocorrect is not hermetic by nature — should we change the `execution_requirements` block to exclude sandbox [like Jin recommends](https://stackoverflow.com/a/50028585)? Honestly saying I failed to find even a single Bazel formatter so I don’t have an inspiration how to deal with this.

Reply:
It's not something I'm familiar with at all (and don't have time to dig into it right now), but I think [this](https://github.com/thundergolfer/bazel-linting-system) repo implements in-place correction (with aspects, which might allow things that regular rules can't do). Probably worth taking a look.

Reply:
Thanks! Seems like it copies source files, then runs format on them and probably copies them back (also I’m not sure anymore). This sounds expensive.

Buildifier provides [an executable rule](https://github.com/bazelbuild/buildtools/blob/master/buildifier/def.bzl). Sounds like it can work since such rules can touch files outside the sandbox (right?) but this means copy-pasting the checking rule definition 😞 

Also found [an interesting comment](https://github.com/bazelbuild/rules_nodejs/issues/570#issuecomment-317223434) that Google doesn’t do this via Bazel (at least three years ago didn’t).

Reply:
> Also found an interesting comment that Google doesn’t do this via Bazel (at least three years ago didn’t).

That's indeed still accurate today - it's also true for linters etc.